apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/extensions/NovaSecurityGroupExtension.java

```diff
@@ -191,14 +191,20 @@ public boolean removeSecurityGroup(String id) {
          return false;
       }
 
-      if (sgApi.get().get(groupId) == null) {
-         return false;
+      // Would be nice to delete the group and invalidate the cache atomically - i.e. use a mutex.
+      // Will make sure that a create operation in parallel won't see inconsistent state.
+
+      boolean deleted = sgApi.get().delete(groupId);
+
+      for (SecurityGroupInRegion cachedSg : groupCreator.asMap().values()) {
```

Comment:
Yes - the security group could've been deleted externally and the local cache still holds a reference to the group. This tries to avoid the situation described in https://issues.apache.org/jira/browse/JCLOUDS-1307.